src/components/NotFound.astro

```diff
@@ -156,3 +156,15 @@ const dictionary = Object.fromEntries(
 	}
 	customElements.define('four-oh-four', FourOhFour);
 </script>
+
+<script is:inline>
+	// Script is inlined to avoid visible flash
+	(() => {
+		// Get current language from URL, then apply it to language selector if matched
+    const cssSelector = `starlight-lang-select option[value="${location.pathname}"]`;
+    const matchedLanguageOption = document.querySelector(cssSelector);
+    if (!!matchedLanguageOption && !!matchedLanguageOption.parentElement) {
```

Comment:
Small detail, but this can be simplified like this:

```suggestion
    if (matchedLanguageOption?.parentElement) {
```

Reply:
Sure thing 👍

https://github.com/withastro/docs/pull/6763/commits/d03a4b97431b66e83ed09be8d5f39e25f4f1146d#diff-3f71d04203961deb2eef1334ac685c4596b67a98a96f0551b33fcf2a454cbd91R167